src/cryptonote_core/blockchain.cpp

```diff
@@ -4552,11 +4565,13 @@ bool Blockchain::handle_block_to_main_chain(const block& bl, const crypto::hash&
     }
   }
 
-  send_miner_notifications(id, already_generated_coins);
+  const crypto::hash seedhash = get_block_id_by_height(crypto::rx_seedheight(new_height));
+  send_miner_notifications(new_height, seedhash, id, already_generated_coins);
 
   for (const auto& notifier: m_block_notifiers)
     notifier(new_height - 1, {std::addressof(bl), 1});
 
+  rx_set_main_seedhash(seedhash.data, std::thread::hardware_concurrency());
```

Comment:
Does this need a version check? It seems to unnecessarily initialize RandomX caches before the RandomX fork height.